pennylane/transforms/qcut.py

```diff
@@ -305,7 +306,15 @@ def fragment_graph(graph: MultiDiGraph) -> Tuple[Tuple[MultiDiGraph], MultiDiGra
             if subgraph.has_node(node2):
                 end_fragment = i
 
-        communication_graph.add_edge(start_fragment, end_fragment, pair=(node1, node2))
+        if start_fragment != end_fragment:
+            communication_graph.add_edge(start_fragment, end_fragment, pair=(node1, node2))
+        else:
+            # The MeasureNode and PrepareNode pair live in the same fragment and did not result
+            # in a disconnection. We can therefore remove these nodes. Note that we do not need
+            # to worry about adding back an edge between the predecessor to node1 and the successor
+            # to node2 because edge connectivity no longer matters in the subgraphs.
```

Comment:
Sorry, maybe this comment doesn't need to change but I'm personally not sure what "edge connectivity no longer matters" means here.

Reply:
I clarified the comment here: b358d1d4f0db5eb183c76581643d1c02a1d4c7be
The idea is that after fragmenting we are happy to convert back to a tape for further processing. Going from graph to tape only requires looking at the nodes and the total order (which is a node attribute), so we don't care about edges in this conversion.